### PR TITLE
Update utils plus old pr

### PR DESCRIPTION
**Pull request by Demonkratiy**
* Updated packages, vulnerabilities removed;
* Removed Jquery usage; 
* Code refactored to use d3.v6, also splitted d3 library on separate packages;
* Include commit from old PR with new tests;

### CODE REVIEW

File: karma.conf.ts [hunk 1]
```diff
@@ -29,17 +29,16 @@
...
     , srcRecursivePath = "lib/**/*.js"
     , srcCssRecursivePath = "lib/**/*.css"
     , coverageFolder = "coverage";
 
 process.env.CHROME_BIN = require("puppeteer").executablePath();
-module.exports = (config: Config) => {
-    config.set(<ConfigOptions>{
+module.exports = (config) => {
+    config.set({
+        mode: "development",
```
Comment (uve) [inline karma.conf.ts:41]: remove it

File: webpack.config.js [hunk 2]
```diff
@@ -30,6 +30,7 @@ const webpack = require("webpack");
 module.exports = {
     entry: './src/index.ts',
     devtool: 'source-map',
+    mode: 'development',
```
Comment (uve) [inline webpack.config.js:33]: remove it

**Review by uve — CHANGES_REQUESTED** [review]

File: webpack.config.js (hunk 2 above)
Comment (Demonkratiy) [reply webpack.config.js]: Was used for debugging tests. Now removed.

File: karma.conf.ts (hunk 1 above)
Comment (Demonkratiy) [reply karma.conf.ts]: Was used for debugging tests. Now removed.

**Review by uve — APPROVED** [review]

**Pull request merged**